home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,101 @@
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
+                            title=umdio_course['name'],
+                            credits=umdio_course['credits'],
+                            description=umdio_course["description"]
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    self._professors(course, semester)
+                    print(course)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        for umdio_professor in umdio_professors:
+            try:
+                professor = Professor.unfiltered.filter(name=umdio_professor['name']).first()
+            except TypeError:
+                continue
+
+            if umdio_professor['name'] == "Instructor: TBA":
+                continue
+
+            if not professor:
+                # To make our lives easier, attempt to automatically verify the professor
+                # following the same criteria in admin.py
+                split_name = umdio_professor['name'].strip().split()
+                first_name = split_name[0].lower().strip()
+                last_name = split_name[-1].lower().strip()
+                query = Professor.verified.filter(
+                    (
+                        Q(name__istartswith=first_name) &
+                        Q(name__iendswith=last_name)
+                    ) |
+                    Q(slug="_".join(reversed(split_name)).lower())
+                )
+
+                professor = Professor(name=umdio_professor['name'], type=Professor.Type.PROFESSOR)
+
+                if not query.exists():
+                    professor.slug = "_".join(reversed(split_name)).lower()
+                    professor.status = Professor.Status.VERIFIED
+
+                professor.save()
+                self.total_num_new_professors += 1
+
+            for entry in umdio_professor['taught']:
+                if entry['course_id'] == course.name and Semester(entry['semester']) == semester:
+                    ProfessorCourse.objects.create(course=course, professor=professor, recent_semester=semester)
+                    break
```

Comment:
why is there a break here? Doesn't this result in a professor only being associated with one course (even if they're teaching multiple) because it breaks after the first result?

Reply:
Not quite. The method that this code resides in is called for every course for every specified semester with the goal of finding all the professors who taught a specific course during a specific semester. The goal of this for loop in particular is to go through the courses a particular professor taught (that is, the professor in the outer for loop not shown here) and see if any of those courses match the semester and course we're currently processing, as provided by the method parameters. If we see that the course and semester we're currently processing matches a course in the courses a professor taught, then we add the new relation to our DB and move onto the next professor, which is done via break. The rest of the courses will be checked when the main function reaches that course and call this method again.

Reply:
got it, so the fact that `_professors` is called with a specific course multiple times is what results in every course being caught here.